src/html/render.ts

```diff
@@ -801,12 +828,27 @@ export async function generateHTMLPage({
 				<meta name="view-transition" content="same-origin"/>
 				<meta name="color-scheme" content="${color_scheme}"/>
 				<meta name="theme-color"/>	
+				${metaScripts}
+				${preloadDependencies ?
+					[...modulePreloadUrls].map(_url => {
+						const url = convertToWebPath(_url);
+						return url.endsWith(".ts")||url.endsWith(".js")||url.endsWith(".tsx")||url.endsWith(".jsx") ? 
+							`<link rel="modulepreload" href="${url}">` : 
+							`<link rel="preload" href="${url}" ${
+								url.endsWith(".css")||url.endsWith(".scss") ? "" : "crossorigin"
+							} as="${
+								url.endsWith(".css")||url.endsWith(".scss") ? "style" : 
+								url.endsWith(".woff2") ? "font" : 
```

Comment:
Should we also check here for other font types like eot ttf? What about preloading of svg?

Reply:
Added woff/ttf
We should encourage the usage of modern web font formats (i.e. woff/woff2). Using the file extension as an indication for the content type is also just a good guess and currently preloading is just enabled for JS modules anyways. Most style sheets don't need preloading because they are directly loaded from the html element. Other dynamic resources like images should/cannot be preloaded like this.